components/browser/session/src/main/java/mozilla/components/browser/session/engine/EngineObserver.kt

```diff
@@ -305,6 +307,85 @@ internal class EngineObserver(
         mediaMap.remove(media)
     }
 
+    override fun onMediaActivated(mediaSessionController: MediaSession.Controller) {
+        store?.dispatch(MediaSessionAction.ActivatedMediaSessionAction(
+            session.id,
+            mediaSessionController
+        ))
+    }
+
+    override fun onMediaDeactivated() {
+        store?.dispatch(MediaSessionAction.DeactivatedMediaSessionAction(session.id))
+    }
+
+    override fun onMediaMetadataChanged(
+        mediaSessionController: MediaSession.Controller,
+        metadata: MediaSession.Metadata
+    ) {
+        store?.dispatch(MediaSessionAction.UpdateMediaMetadataAction(
+            session.id,
+            mediaSessionController,
+            metadata
+        ))
+    }
+
+    override fun onMediaPlaybackStateChanged(
+        mediaSessionController: MediaSession.Controller,
+        playbackState: MediaSession.PlaybackState
+    ) {
+        store?.dispatch(MediaSessionAction.UpdateMediaPlaybackStateAction(
+            session.id,
+            mediaSessionController,
+            playbackState
```

Comment:
Do we need to pass the `mediaSessionController` along every time? Does it change after the activate call?

Reply:
I'm assuming the controller does not change after activate.  I'll double check.

Reply:
Confirmed, no change is expected.  I'll remove passing in `mediaSessionController` after activate.